### PR TITLE
workflow: align imitation warm-start env contract

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Aligns the issue 749 / issue 1108 imitation warm-start pipeline around a shared environment contract so trajectory collection, BC pretraining, and PPO fine-tuning use compatible scenario/env metadata after the latest `main` imitation-contract changes.

This is an unblocker PR related to issue 1108. It does not complete issue 1108 because the full SLURM experiment artifacts, durable pointers, and policy-analysis comparison are still outstanding.

## Linked Issues
- Relates to issue 1108
- Relates to issue 749

## What Changed
- Added the Auxme launcher for the issue-1108 BC warm-start PPO chain.
- Extended imitation configs with scenario/env contract metadata and CPU BC device control for the Auxme CUDA mismatch.
- Threaded training-config env overrides/factory kwargs through `scripts/training/imitation_env_contract.py` and the collection/BC/fine-tune scripts.
- Recorded the execution history and remaining evidence boundary in `docs/context/issue_1108_bc_warm_start_execution.md`.
- Added/updated targeted integration coverage for the BC device path and warm-start config loading.

## Why It Matters
- Added value: removes the observation-space/device mismatch that blocked the warm-start chain before the actual long-running PPO evidence can be trusted.
- Expected impact: makes the pipeline fail on real experiment outcomes rather than on avoidable collector/BC/fine-tune contract drift.
- Why this is worth merging now: it preserves a validated, latest-main-compatible unblocker while issue 1108 remains a SLURM/artifact follow-up.

## Validation / Proof
- Commands run:
  - `uv run python -m py_compile robot_sf/training/imitation_config.py scripts/training/imitation_env_contract.py scripts/training/collect_expert_trajectories.py scripts/training/pretrain_from_expert.py scripts/training/train_ppo_with_pretrained_policy.py`
  - `bash -n SLURM/Auxme/issue_1108_bc_warm_start.sl`
  - `uv run ruff check robot_sf/training/imitation_config.py scripts/training/imitation_env_contract.py scripts/training/collect_expert_trajectories.py scripts/training/pretrain_from_expert.py scripts/training/train_ppo_with_pretrained_policy.py tests/training/test_imitation_env_contract.py tests/training/test_pretrain_from_expert.py tests/training/test_train_ppo_with_pretrained_policy.py tests/integration/test_ppo_pretraining_pipeline.py`
  - `uv run ruff format --check robot_sf/training/imitation_config.py scripts/training/imitation_env_contract.py scripts/training/collect_expert_trajectories.py scripts/training/pretrain_from_expert.py scripts/training/train_ppo_with_pretrained_policy.py tests/training/test_imitation_env_contract.py tests/training/test_pretrain_from_expert.py tests/training/test_train_ppo_with_pretrained_policy.py tests/integration/test_ppo_pretraining_pipeline.py`
  - `uv run pytest tests/training/test_imitation_env_contract.py tests/training/test_pretrain_from_expert.py tests/training/test_train_ppo_with_pretrained_policy.py tests/integration/test_ppo_pretraining_pipeline.py -q`
  - `BASE_REF=origin/main scripts/dev/check_docs_proof_consistency_diff.sh`
  - `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main DISPLAY= MPLBACKEND=Agg SDL_VIDEODRIVER=dummy scripts/dev/pr_ready_check.sh`
- Evidence that the change works here:
  - targeted pytest: `22 passed`
  - final readiness gate: `3627 passed, 10 skipped, 3 warnings in 429.49s`
  - doc/proof consistency: passed for 12 changed files
  - touched-definition docstring TODO check: clean
- Benchmarks or smoke tests, if applicable:
  - No benchmark-strength SLURM rerun from this host. `local.machine.md` has `allow_slurm_submission: false`.

## Risks / Rollout
- Compatibility risks: config fields are additive and default to current behavior when omitted.
- Failure modes: the actual experiment can still fail on model quality, long-run training stability, or artifact publication after this unblocker.
- Rollback or fallback plan: revert this PR and keep the experiment blocked on the prior collector/BC/PPO contract mismatch.

## Docs / Provenance
- Updated docs:
  - `docs/context/README.md`
  - `docs/context/issue_1108_bc_warm_start_execution.md`
  - `docs/context/issue_749_bc_preinit_ppo_launch_packet.md`
- Relevant design or provenance notes: `docs/context/issue_1108_bc_warm_start_execution.md` records the earlier SLURM attempts and explicitly separates local cache output from durable evidence.
- Any assumptions that need to be preserved: worktree-local `output/` coverage files are disposable; issue completion still requires durable dataset/checkpoint pointers plus policy-analysis comparison against `27dbe5xu` and `b60iopxt`.

## Follow-Up Issues
- Deferred work: complete the experiment on a machine allowed to submit/inspect Auxme SLURM jobs, then publish durable artifacts and analysis results.
- Issues opened for follow-up: none; issue 1108 remains open for the deferred experiment evidence.

## Reviewer Notes
- Please verify closely that the collector, BC pretrain, and PPO fine-tune scripts all resolve the same scenario/env contract.
- Known limitation: this PR does not claim benchmark-strength experiment completion.

## Summary by CodeRabbit

* **New Features**
  * Added automated SLURM job script for BC warm-start PPO training pipelines
  * Training configurations now support environment-specific overrides and factory settings for flexible scenario and environment setup

* **Documentation**
  * Added comprehensive documentation for BC warm-start PPO execution workflow and configuration requirements

* **Tests**
  * Added integration tests validating warm-start configuration consistency across training stages

[Review Change Stack](https://app.coderabbit.ai/change-stack/ll7/robot_sf_ll7/pull/1307?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: scripts/training/collect_expert_trajectories.py [hunk 1]
```diff
@@ -37,7 +38,6 @@
     select_scenario,
 )
 from scripts.training.imitation_env_contract import (
-    load_training_env_overrides,
     observation_contract_from_space,
 )
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/training/collect_expert_trajectories.py:43]: Fixed. `collect_expert_trajectories.py` now imports and uses `load_training_env_overrides` and `load_training_env_factory_kwargs` from `imitation_env_contract.py` instead of a local YAML loader.

File: scripts/training/collect_expert_trajectories.py [hunk 2]
```diff
@@ -75,12 +75,40 @@ def _resolve_scenario_config(
...
+    if config_path is None:
+        return {}, {}
+    raw = yaml.safe_load(config_path.read_text(encoding="utf-8")) or {}
+    env_overrides = raw.get("env_overrides") or {}
+    env_factory_kwargs = raw.get("env_factory_kwargs") or {}
+    if not isinstance(env_overrides, RuntimeMapping):
+        raise TypeError(f"{config_path}: env_overrides must be a mapping")
+    if not isinstance(env_factory_kwargs, RuntimeMapping):
+        raise TypeError(f"{config_path}: env_factory_kwargs must be a mapping")
+    return dict(env_overrides), dict(env_factory_kwargs)
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/training/collect_expert_trajectories.py]: Fixed. Removed the redundant `_load_env_contract` helper and its `RuntimeMapping` import; loading now goes through the shared helpers.

File: scripts/training/collect_expert_trajectories.py [hunk 3]
```diff
@@ -321,7 +356,8 @@ def main(argv: Sequence[str] | None = None) -> int:
 
     scenario_path = _resolve_scenario_config(args.scenario_config, args.training_config)
     training_config_path = args.training_config.resolve() if args.training_config else None
-    env_overrides = load_training_env_overrides(training_config_path)
+    env_contract_path = args.env_config or args.training_config
+    env_overrides, env_factory_kwargs = _load_env_contract(env_contract_path)
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/training/collect_expert_trajectories.py]: Fixed. Collector loading now explicitly applies `training_config` first, then overlays `env_config` values when present. Added `test_collect_trajectories_merges_training_then_env_contract` to lock the precedence and provenance metadata.

File: scripts/training/imitation_env_contract.py [hunk 4]
```diff
@@ -99,8 +114,13 @@ def make_training_contract_env(
...
-    _apply_env_overrides(env_config, load_training_env_overrides(training_config_path))
-    env = make_robot_env(config=env_config, seed=seed)
+    resolved_overrides = load_training_env_overrides(training_config_path)
+    resolved_overrides.update(env_overrides or {})
+    resolved_factory_kwargs = load_training_env_factory_kwargs(training_config_path)
+    resolved_factory_kwargs.update(env_factory_kwargs or {})
+
+    _apply_env_overrides(env_config, resolved_overrides)
+    env = make_robot_env(config=env_config, seed=seed, **resolved_factory_kwargs)
     if observation_keys:
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/training/imitation_env_contract.py:131]: Fixed. `make_training_contract_env` now rejects `config` and `seed` in resolved `env_factory_kwargs` before calling `make_robot_env`, with a targeted regression test.

**Comment by ll7** [comment]
Autonomous review complete for #1307.

Review result:
- Treated this as a scoped unblocker for the #1108/#749 warm-start pipeline, not as completion of #1108.
- The PR consistently threads env contract metadata through trajectory collection, BC pretraining, and PPO fine-tuning.
- Collector config precedence is explicit: `--training-config` first, then `--env-config` overlays.
- `make_training_contract_env` fails fast on reserved factory kwargs (`config`, `seed`) before calling `make_robot_env`.
- The docs correctly keep local `output/`/SLURM products as non-evidence until durable dataset/checkpoint pointers and policy-analysis comparison exist.
- All visible review threads are resolved.

Validation evidence I ran locally:
- `uv run python -m py_compile robot_sf/training/imitation_config.py scripts/training/imitation_env_contract.py scripts/training/collect_expert_trajectories.py scripts/training/pretrain_from_expert.py scripts/training/train_ppo_with_pretrained_policy.py` -> passed.
- `bash -n SLURM/Auxme/issue_1108_bc_warm_start.sl` -> passed.
- Ruff check and Ruff format check passed on touched training scripts/tests.
- Targeted training/imitation tests: `uv run pytest tests/training/test_imitation_env_contract.py tests/training/test_pretrain_from_expert.py tests/training/test_train_ppo_with_pretrained_policy.py tests/integration/test_ppo_pretraining_pipeline.py -q` -> `24 passed`.
- `BASE_REF=origin/main scripts/dev/check_docs_proof_consistency_diff.sh && git diff --check origin/main...HEAD` -> passed.
- Full readiness: `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main DISPLAY= MPLBACKEND=Agg SDL_VIDEODRIVER=dummy scripts/dev/pr_ready_check.sh` -> `3629 passed, 10 skipped, 3 warnings`.
- GitHub checks are green and the PR is mergeable/clean.

Artifact decision:
- Ignored local validation artifacts (`.venv`, caches, `output/coverage`) are disposable.
- This PR intentionally does not claim benchmark-strength SLURM or policy-analysis evidence; issue #1108 remains open for durable experiment artifacts and comparison results.

Added `merge-ready`.
